File: 01gpio/03digital_in/digital_input_pud_wpi.js
const wpi = require('node-wiring-pi');

const PIN_BTN = 23;
let [val, old] = [-1, -1];

wpi.wiringPiSetupGpio();
wpi.pinMode(PIN_BTN, wpi.INPUT);
wpi.pullUpDnControl(PIN_BTN, wpi.PUD_DOWN);

while(true){
	val = wpi.digitalRead(PIN_BTN);
	if(val != old){
		old = val;
		console.log(val);
	}
}

File: 01gpio/03digital_in/digital_input_pud_wpi.py
import wiringpi as wpi

PIN = 18
wpi.wiringPiSetupGpio()
wpi.pinMode(PIN, wpi.INPUT)
wpi.pullUpDnControl(PIN, wpi.PUD_UP)
#wpi.pullUpDnControl(PIN, wpi.PUD_DOWN)
old_val = -1
while True:
	val = wpi.digitalRead(PIN)
	if val != old_val:
		old_val = val
		print(val)
